app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -2294,16 +2311,67 @@ public void exportData(Function<Entity, Void> handler) throws RegistryStorageExc
     }
 
     /**
-     * @see RegistryStorage#importData(io.apicurio.registry.storage.impexp.EntityInputStream)
+     * @see RegistryStorage#importData(io.apicurio.registry.storage.impexp.EntityInputStream, boolean, boolean)
      */
     @Override
     @Transactional
-    public void importData(EntityInputStream entities) throws RegistryStorageException {
-        handles.withHandleNoException( handle -> {
+    public void importData(EntityInputStream entities, boolean preserveGlobalId, boolean preserveContentId) throws RegistryStorageException {
+        Map<Long, Long> contentIdMapping = new HashMap<>();
+        // To handle issue, when we do not know new content id before importing the actual content
+        ArrayList<ArtifactVersionEntity> waitingForContent = new ArrayList<>();
+
+        handles.withHandleNoException(handle -> {
             Entity entity = null;
-            while ( (entity = entities.nextEntity()) != null ) {
-                if (entity != null) {
-                    importEntity(handle, entity);
+            while ((entity = entities.nextEntity()) != null) {
+                if (entity.getEntityType() == EntityType.ArtifactVersion) {
```

Comment:
I recommend putting the `ArtifactVersion` logic below the `Content` logic.  It should make the code easier to read/interpret in the future.  Logically we would expect to process the Content entities before the artifact version entities.  But even if we don't (in practice), which should **think** about it like we do.